lib/customerio/client.rb

```diff
@@ -110,6 +110,20 @@ def remove_from_segment(segment_id, customer_ids)
       }))
     end
 
+    def track_push_notification_open(attributes = {})
+        keys = [:delivery_id, :device_id, :timestamp]
+        attributes = Hash[attributes.map { |(k,v)| [ k.to_sym, v ] }].
+            select { |k, v| keys.include?(k) }
+
+        raise ParamError.new('delivery_id must be a non-empty string') unless attributes[:delivery_id] != "" and !attributes[:delivery_id].nil?
+        raise ParamError.new('device_id must be a non-empty string') unless attributes[:device_id] != "" and !attributes[:device_id].nil?
+        raise ParamError.new('timestamp must be a valid timestamp') unless valid_timestamp?(attributes[:timestamp])
+
+        verify_response(request(:post, track_push_notification_open_path, attributes.merge({
+            :event => 'opened'
```

Comment:
This is the only valid value for `event`, according to the integration guide - figured it should just be hard-coded.

Reply:
`opened`, `converted`, `delivered` [should all be valid events](https://customer.io/docs/api/#operation/pushMetrics). I think it would make sense to have methods for all of those events or a generic method such as `track_push_notification_event(event_name:, attributes: {})` that can be used for everything.

I'll work on that change and get this merged in an upcoming release.